src/main/java/com/example/bookstore/service/shoppingcart/ShoppingCartServiceImpl.java

```diff
@@ -54,10 +66,27 @@ public ShoppingCartDto updateQuantity(Long id,
         return getShoppingCart();
     }
 
+    @Transactional
+    @Override
+    public ShoppingCart getShoppingCartModel() {
+        User user = userService.getUser();
+        return shoppingCartRepository.findByUserId(user.getId()).orElseThrow(()
+                -> new EntityNotFoundException("Can't find cart by id: " + user.getId()));
+    }
+
+    @Transactional
+    @Override
+    public void confirmPurchase(ShoppingCart shoppingCart) {
```

Comment:
confirming purchase consists only of deleting shopping cart and creating new one?